topic.go

```diff
@@ -31,6 +31,46 @@ func (t *Topic) String() string {
 	return t.topic
 }
 
+// SetScoreParams sets the topic score parameters if the pubsub router supports peer
+// scoring
+func (t *Topic) SetScoreParams(p *TopicScoreParams) error {
```

Comment:
Can `SetScoreParameters` be called at any time during runtime to reset the topic's score parameters?

* If yes, how does this interact with any existing cached scores for peers in that topic? Should we be resetting them? Or at least letting the user decide through a boolean param?
* If no, we should remove that gotcha by erroring if the user attempts to change the parameters _after_ the first publish or subscribe operation.

Reply:
Well, as it is it can be called at any time; but we can change it to check weather parameters already exist for the score and if that's the case we can error out.
This sounds like more reasonable behaviour.

Reply:
Added check for already existing parameters and error if that's the case.

Reply:
If they can only be set a topic construction time, why can't we make it a `TopicOpt`?

Reply:
we could, but there is the issue of Join/Leave/Join pattern.

Reply:
Also, for the sake of argument, there is currently no such thing as a TopicOpt; we'd have to add it and definite semantics for it.